snuba/datasets/schemas/join.py

```diff
@@ -66,13 +69,16 @@ class TableJoinNode(JoinNode):
     It can be a table or a view.
     """
     alias: str
-    schema: TableSchema
+    table: TableSource
 
     def __str__(self) -> str:
-        return f"{self.schema.get_data_source()} {self.alias}"
+        return f"{self.table.format_from()} {self.alias}"
+
+    def get_tables(self) -> Mapping[str, TableSource]:
+        return {self.alias: self.table}
 
-    def get_schemas(self) -> Mapping[str, Schema]:
-        return {self.alias: self.schema}
+    def get_columns(self) -> ColumnSet:
+        raise self.table.get_columns()
```

Comment:
I don't think this is what you meant to put here.

Out of curiosity, I did look to see if mypy would allow a method with a return value that is impossible to actually reach — apparently it does, but it did raise this error:

```
snuba/datasets/schemas/join.py:81: error: Exception must be derived from BaseException
```

Reply:
Yeah, this is a codemod issue.
I need to dig into why this did not make this test fail:
> tests/datasets/schemas/test_join_schema.py:    columns = schema.get_columns()

In the other cases I can see this not being used in a join since it is used mostly by: DDL, replacers, native query driver, some cases of column_expr. None of which is ever used by the join dataset.

About mypy, that makes sense since I think I forgot doing a --strict pass after the last code mod (which included this change), thus I missed the error.

Will fix

Reply:
This issue  made me realize that abstracting TableJoinNode as a JoinNode that contains a TableSource is not sound. TableJoinNode is a TableSource with an alias that supports JoinNode features. Making TableJoinNode extend TableSource makes the isinstance calls in groups.py not needed.